Gpio driver - Added a program to blink led in raspberry pi. "gpio-driver" branched from "cmake-make" branch.

afraz-ashik opened the pull request.

afraz-ashik committed:
Added a sample makefile.

afraz-ashik committed:
Updated the test makefile.

afraz-ashik committed:
Modified the makefile according to requirements.

afraz-ashik committed:
Added debug target

afraz-ashik committed:
Use of static pattern rule and wildcards.

afraz-ashik committed:
Use of "+=" for variables.
made target "linux" a phony target.

afraz-ashik committed:
Logic Correction in line 46.

afraz-ashik committed:
Created a file CMakeLists.txt for cmake. Removed .txt from gitignore file.

afraz-ashik committed:
Updated CMakeLists.txt

afraz-ashik committed:
Updated the CMakeLists.txt file. Updated the gitignore file

afraz-ashik committed:
Updated the Debug and Release folder directory.
Updated gitignore file

afraz-ashik committed:
added toolchain feature for building rpi

afraz-ashik committed:
Used externalprojects

afraz-ashik committed:
Changed Output names

afraz-ashik committed:
Renamed Output names in projects

afraz-ashik committed:
Added a conditional macro for native building and building for raspberrypi.
in cmakelist for arm code added and linked necessary libraries. Added libgpiod library for gpio configuration.

afraz-ashik committed:
renamed target

afraz-ashik committed:
Updated readme

afraz-ashik committed:
static analyzer flag added.

afraz-ashik committed:
removed unnecessary flag.

afraz-ashik committed:
Spilt function in ledSimulation.c . Reduced size of functions.
Added a static global variable.

afraz-ashik committed:
changed return statement in line 86

afraz-ashik committed:
In-built header file called from user-defined header file

afraz-ashik committed:
Newline added.  comments updated.

afraz-ashik committed:
Created folder gpiodWrapper with corresponding source and header files. Updated comments. Updated cmake include directories with wrapper folder.

afraz-ashik committed:
Removed repetitions.
Added newlines.

afraz-ashik committed:
Modified Error checks in gpiodWrapper source file

afraz-ashik committed:
Changed void functions to bool type.

afraz-ashik committed:
Renamed files,functions and folders.
Used Pass by reference to make the code efficient.
Upadeted the cmake files and make files.

afraz-ashik committed:
Updated Readme.

afraz-ashik committed:
Updated variables names in makefile.

afraz-ashik committed:
added else condition for ifdef

afraz-ashik committed:
Added spaces.

afraz-ashik committed:
Reomved unneccessary lines of code.

adarshtrenser2023 commented on LedSimulation/LedSimulation.c:
```diff
@@ -12,45 +13,88 @@
+//*****************************************************************************
+bool LedSimulationBlinkLED(struct gpiod_line **pstLine)
+{
+    (void) *pstLine;
```
what is the use of line 38

adarshtrenser2023 commented on LedSimulation/LedSimulation.c:
```diff
@@ -12,45 +13,88 @@
 //***************************** Local Constants *******************************
 
 //***************************** Local Variables *******************************
+static bool gsblLedStatus = false;
```
since there is only one function using this function you can use this locally why changed it from local to global?

afraz-ashik replied:
that variable was needed across both functions in that file. I've changed the scope of that variable and have now declared that variable inside main instead.

adarshtrenser2023 commented on gpiodTools/gpiodTools.c:
```diff
@@ -0,0 +1,108 @@
+// Return  : True.
+// Notes   : None.
+//*****************************************************************************
+bool gpiodToolsGpioClose(struct gpiod_line **pstLine, 
```
does pstLine need double pointer..? cant we use it without double pointer?
also if it is double pointer you need to keep pp as prefix

adarshtrenser2023 commented on gpiodTools/gpiodTools.c:
```diff
@@ -0,0 +1,108 @@
+
+    if (NULL == pstLine || NULL == pstChip)
+    {
+        printf("NULL!");
```
what is null? print statement should be meaningful..please check all print statements

afraz-ashik committed:
declared variabel in main and passed to ledSimulation function.
Proper print statements for error conditions.
Added proper prefix.
Avoided use of global variable.

afraz-ashik committed:
Proper check conditions in line 47 and 56 of ledSimulation.c

afraz-ashik committed:
Revert "Proper check conditions in line 44, 47 and 56 of ledSimulation.c"

This reverts commit 5d8adef47c26d6e48cdaf6b7f3989045d4c58dba.

afraz-ashik committed:
Proper bool check condtions in LedSimulation.c

afraz-ashik committed:
Proper tabs added.

afraz-ashik committed:
Avoided use of double pointer in ledSimulation. Added else case for typecast.

afraz-ashik committed:
Changed print statement in gpiodTools.c:42

afraz-ashik commented on gpiodTools/gpiodTools.c:
```diff
@@ -0,0 +1,123 @@
+
+        if (NULL == ppstLine || NULL == ppstChip)
+        {
+            perror("Passed pointers are NULL!");
```
changed the statement as suggested.

afraz-ashik commented on LedSimulation/LedSimulation.c:
```diff
@@ -12,45 +13,92 @@
 //******************************* Include Files *******************************
 #include "LedSimulation.h"
 
+
```
Removed newlines.
